File: app/src/main/java/me/imzack/app/end/view/contract/AboutViewContract.java
package me.imzack.app.end.view.contract;

import me.imzack.app.end.view.adapter.AboutPagerAdapter;

public interface AboutViewContract extends BaseViewContract {

    void showInitialView(String versionName, AboutPagerAdapter aboutPagerAdapter);

    void onAppBarScrolled(float headerLayoutAlpha);

    void onAppBarScrolledToCriticalPoint(String toolbarTitle);

    void translateViewWhenIncline(boolean shouldTranslateX, float translationX, boolean shouldTranslateY, float translationY);

    void resetViewTranslation();

    void backToTop();

    void pressBack();
}

File: app/src/main/kotlin/me/imzack/app/end/view/contract/AboutViewContract.kt
package me.imzack.app.end.view.contract

import me.imzack.app.end.view.adapter.AboutPagerAdapter

interface AboutViewContract : BaseViewContract {

    fun showInitialView(versionName: String, aboutPagerAdapter: AboutPagerAdapter)

    fun onAppBarScrolled(headerLayoutAlpha: Float)

    fun onAppBarScrolledToCriticalPoint(toolbarTitle: String)

    fun translateViewWhenIncline(shouldTranslateX: Boolean, translationX: Float, shouldTranslateY: Boolean, translationY: Float)

    fun resetViewTranslation()

    fun backToTop()

    fun pressBack()
}
